client/testsuites/v1/testsuites.go

```diff
@@ -107,7 +146,108 @@ func (s TestSuitesClient) DeleteAll() error {
 	u.SetKind("TestSuite")
 	u.SetAPIVersion("tests.testkube.io/v1")
 	err := s.Client.DeleteAllOf(context.Background(), u, client.InNamespace(s.Namespace))
-	return err
+	if err != nil {
+		return err
+	}
+
+	u = &unstructured.Unstructured{}
+	u.SetKind("Secret")
+	u.SetAPIVersion("v1")
+	u.SetLabels(testsuiteSecretDefaultLabels)
+
+	return s.Client.DeleteAllOf(context.Background(), u, client.InNamespace(s.Namespace))
+}
+
+// CreateTestSecrets creates corresponding test vars secrets
+func (s TestSuitesClient) CreateTestsuiteSecrets(testsuite *testsuitev1.TestSuite) error {
+	secretName := secretName(testsuite.Name)
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      secretName,
+			Namespace: s.Namespace,
+			Labels:    testsuiteSecretDefaultLabels,
+		},
+		Type:       corev1.SecretTypeOpaque,
+		StringData: map[string]string{},
+	}
+
+	testVarsToSecret(testsuite, secret)
+
+	if len(secret.StringData) > 0 {
+		err := s.Client.Create(context.Background(), secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s TestSuitesClient) UpdateTestsuiteSecrets(testsuite *testsuitev1.TestSuite) error {
+	secret, err := s.LoadTestVariablesSecret(testsuite)
+	if err != nil {
+		return err
+	}
+
+	testVarsToSecret(testsuite, secret)
+
+	if len(secret.StringData) > 0 {
+		err := s.Client.Update(context.Background(), secret)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (s TestSuitesClient) LoadTestVariablesSecret(testsuite *testsuitev1.TestSuite) (*corev1.Secret, error) {
+	secret := &corev1.Secret{}
+	err := s.Client.Get(context.Background(), client.ObjectKey{Namespace: s.Namespace, Name: secretName(testsuite.Name)}, secret)
+	return secret, err
+}
+
+// testVarsToSecret loads secrets data passed into Test CRD and remove plain text data
+func testVarsToSecret(testsuite *testsuitev1.TestSuite, secret *corev1.Secret) {
```

Comment:
can we have an interface and remove duplication?

Reply:
Was thinking about it but decided to simplify usage of particular resource objects - but idea about removing dups is good - maybe will add type based on some common data structure